### PR TITLE
Fix builds of modpack on M1 Mac

### PR DESCRIPTION
**Pull request by austintraver**
Previous error:

```
error[E0308]: mismatched types=======> ] 190/191: training_modpack                               
   --> src/common/events.rs:183:34
    |
183 |         std::ffi::CStr::from_ptr(smash_version.name.as_ptr() as *const i8)
    |         ------------------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const u8`, found `*const i8`
    |         |
    |         arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`
note: associated function defined here
   --> /Users/austin/.cargo/skyline/toolchain/skyline/lib/rustlib/src/rust/library/core/src/ffi/c_str.rs:260:25
```

Using `ffi::c_char` directly fixes this compilation error.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -44,19 +45,18 @@ jobs:
...
+          target: x86_64-unknown-linux-gnu
+      - name: Install Skyline
+        run: |
+          cargo install cargo-skyline
+          cargo-skyline skyline update-std
       - name: Build release NRO
         id: build_release
         run: cargo-skyline skyline build --release
-        env:
-          HOME: /root
```
Comment (austintraver) [inline .github/workflows/rust.yml:59]: Wasn't able to get the build to succeed without removing this line

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -44,19 +45,18 @@ jobs:
...
     runs-on: ubuntu-latest
-    container:
-      image: jugeeya/cargo-skyline:3.2.0-no-dkp
     steps:
       - uses: actions/checkout@v3
-      - uses: Swatinem/rust-cache@v2
-        name: Rust Cache
+      - uses: dtolnay/rust-toolchain@nightly
         with:
-          prefix-key: "plugin"
```
Comment (austintraver) [inline .github/workflows/rust.yml:54]: I found that using a fresh toolchain caused the build to succeed

**Comment by austintraver** [comment]
This PR has a bit of scope creep, as there were issues not only in getting the build to succeed locally, but getting it to succeed when running on the GitHub Actions Runner as well

**Review by jugeeya — APPROVED** [review]
LGTM

**Pull request merged**